cpp/examples/graph_partitioning/graph_partition_examples.cpp

```diff
@@ -0,0 +1,250 @@
+      std::string("vertices@rank_").append(std::to_string(comm_rank)).c_str();
+    raft::print_device_vector(
+      vertex_paritition_title, (*renumber_map).data(), (*renumber_map).size(), std::cout);
+  }
+
+  std::vector<vertex_t> h_vertices_in_this_proces((*renumber_map).size());
+
+  raft::update_host(h_vertices_in_this_proces.data(),
+                    (*renumber_map).data(),
+                    (*renumber_map).size(),
+                    handle.get_stream());
+  handle.sync_stream();
+
+  assert(size_of_the_vetex_partition_assigned_to_this_process == (*renumber_map).size());
+
+  // The position of a vertex in the `renumber_map` is indicative of its new (aka renumberd) vertex
+  // id
+
+  // The new (aka renumbed) id of the first vertex, ie the vertex at position 0 of `renumber_map`,
+  // assigned to this process
+
+  vertex_t renumber_vertex_id_of_local_first = graph_view.local_vertex_partition_range_first();
+
+  // The new (aka renumbed) id of the last vertex, ie the vertex at position 0 of `renumber_map`,
+  // assigned to this process
+
+  vertex_t renumber_vertex_id_of_local_last = graph_view.local_vertex_partition_range_last();
+
+  if (renumber_map) {
+    thrust::for_each(thrust::host,
+                     thrust::make_zip_iterator(thrust::make_tuple(
+                       h_vertices_in_this_proces.begin(),
+                       thrust::make_counting_iterator(renumber_vertex_id_of_local_first))),
+                     thrust::make_zip_iterator(thrust::make_tuple(
+                       h_vertices_in_this_proces.end(),
+                       thrust::make_counting_iterator(renumber_vertex_id_of_local_last))),
+                     [comm_rank](auto old_and_new_id_pair) {
+                       auto old_id = thrust::get<0>(old_and_new_id_pair);
+                       auto new_id = thrust::get<1>(old_and_new_id_pair);
+                       printf("owned by rank %d, original vertex id %d --- becomes -->  %d\n",
+                              comm_rank,
+                              static_cast<int>(old_id),
+                              static_cast<int>(new_id));
+                     });
+  }
+  // Look into edge partitions and their associated edge properties (if any)
+  // Non-owning of the edge edge_weights object
+  auto edge_weight_view = edge_weights ? std::make_optional((*edge_weights).view()) : std::nullopt;
+
+  for (size_t ep_idx = 0; ep_idx < graph_view.number_of_local_edge_partitions(); ++ep_idx) {
+    // Toplogy
+    auto edge_partition = graph_view.local_edge_partition_view(ep_idx);
+
+    auto number_of_edges = edge_partition.number_of_edges();
+    auto offsets         = edge_partition.offsets();
+    auto indices         = edge_partition.indices();
+
+    auto offsets_title = std::string("offsets_")
+                           .append(std::to_string(comm_rank))
+                           .append("_")
+                           .append(std::to_string(ep_idx))
+                           .c_str();
```

Comment:
Can you do this?  Doesn't the string go out of scope once the line is completed?

I feel like you need to leave off the `c_str` call here, and add it to line 197 where you use it in order to guarantee that the memory is not reused.